airflow/contrib/auth/backends/ldap_auth.py

```diff
@@ -55,16 +55,20 @@ class LdapException(Exception):
 
 
 def get_ldap_connection(dn=None, password=None):
-    tls_configuration = None
-    use_ssl = False
+    cacert = None
     try:
         cacert = configuration.conf.get("ldap", "cacert")
-        tls_configuration = Tls(validate=ssl.CERT_REQUIRED, ca_certs_file=cacert)
-        use_ssl = True
-    except Exception:
+    except AirflowConfigException:
         pass
 
-    server = Server(configuration.conf.get("ldap", "uri"), use_ssl, tls_configuration)
+    tls_configuration = Tls(validate=ssl.CERT_REQUIRED,
+                            version=ssl.PROTOCOL_SSLv23,
```

Comment:
Do we really want to say SSL v2? Isn't that a security risk? Default to TLSv1.2 with a config option to switch back?

Reply:
I agree with this. We should definitely default to `TLSv1.2`. We underwent a pentest for one of our environment, and this was raised as an issue.

Reply:
I've looked at the code of ldap3 and if we don't specify a version then it uses `ssl.create_default_context()`, the doc string for which says:

> Return a new SSLContext object with default settings for the given purpose. The settings are chosen by the ssl module, and usually represent a higher security level than when calling the SSLContext constructor directly.

So we should not specify any version.

Reply:
I.probably misread the docs, but I'm not sure if you are looking at the right place either. From 

The parameter ssl_version specifies which version of the SSL protocol to use. Typically, the server chooses a particular protocol version, and the client must adapt to the server’s choice. Most of the versions are not interoperable with the other versions. If not specified, the default is PROTOCOL_SSLv23; it provides the most compatibility with other versions.

Reply:
Elsewhere in that doc:

> PROTOCOL_SSLv23
>
>  Alias for PROTOCOL_TLS

The wording I quoted above is from https://docs.python.org/2/library/ssl.html#ssl.create_default_context and is called here https://github.com/cannatag/ldap3/blob/master/ldap3/core/tls.py#L172-L187 - it still sets the validate flag.